clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -440,6 +442,16 @@ public boolean isConnectionSetupTimeout(String id, long now) {
         return now - lastConnectAttemptMs(id) > connectionSetupTimeoutMs(id);
     }
 
+    /**
+     * Return the Set of nodes whose connection setup has timed out.
+     * @param now the current time in ms
+     */
+    public Set<String> nodesWithConnectionSetupTimeout(long now) {
```

Comment:
Do we need a `Set` here? Not sure we need to pay the cost of creating the `HashSet`, etc.